### ISSUE TITLE
Dagster integration throws exception

### ISSUE THREAD

**Issue by elic-eon** [OpenLineage/OpenLineage#2043]
### Problem
The Dagster sensor seems not compatible with the latest Dagster (1.4.4) and I believed there is a [breaking change](https://github.com/dagster-io/dagster/blob/master/CHANGES.md#event-records) since Dagster 0.15.

Here is the full error log:
```
2023-08-10 12:19:16 +0800 - dagster.daemon.SensorDaemon - ERROR - Sensor daemon caught an error for sensor openlineage_sensor
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_daemon/sensor.py", line 536, in _process_tick_generator
    yield from _evaluate_sensor(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_daemon/sensor.py", line 672, in _evaluate_sensor
    sensor_runtime_data = code_location.get_external_sensor_execution_data(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_core/host_representation/code_location.py", line 845, in get_external_sensor_execution_data
    return sync_get_external_sensor_execution_data_grpc(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_api/snapshot_sensor.py", line 78, in sync_get_external_sensor_execution_data_grpc
    raise DagsterUserCodeProcessError.from_error_info(result.error)
dagster._core.errors.DagsterUserCodeProcessError: dagster._core.errors.SensorExecutionError: Error occurred during the execution of evaluation_fn for sensor openlineage_sensor

Stack Trace:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_grpc/impl.py", line 363, in get_external_sensor_execution
    with user_code_error_boundary(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_core/errors.py", line 296, in user_code_error_boundary
    raise error_cls(

The above exception was caused by the following exception:
TypeError: EventRecordsFilter.__new__() missing 1 required positional argument: 'event_type'

Stack Trace:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_core/errors.py", line 289, in user_code_error_boundary
    yield
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_grpc/impl.py", line 369, in get_external_sensor_execution
    return sensor_def.evaluate_tick(sensor_context)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_core/definitions/sensor_definition.py", line 722, in evaluate_tick
    result = self._evaluation_fn(context)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/dagster/_core/definitions/sensor_definition.py", line 1018, in _wrapped_fn
    result.append(next(raw_evaluation_result))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/openlineage/dagster/sensor.py", line 65, in _openlineage_sensor
    event_log_records = get_event_log_records(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/openlineage/dagster/utils.py", line 46, in get_event_log_records
    EventRecordsFilter(
```

**Comment by mobuchowski** [comment]
@elic-eon you're right... Dagster integration unfortunately has no active maintainer. Current ones - like me - only fix errors that are caused by other OpenLineage library changes. 

If you can contribute fix to that and make it work, we'll review it.

**Comment by tsungchih** [comment]
@mobuchowski After having studied a little bit further, I would like to give it a try to contribute fix. Now I can have all tests passed with Dagster 1.5.5. I'm trying to figure out how to create a PR.

**Comment by JDarDagran** [comment]
@tsungchih that sounds great! Please let us know if you need any help. You can also reach out on our Slack, you're very welcome 🙂

**Comment by tsungchih** [comment]
@JDarDagran I saw `ci/circleci: unit-test-integration-dagster` is failed. After having studied in advance, I found there are some issue related to the Dagster API changes especially spanned over many versions. May I ask which Dagster version the `ci/circleci: unit-test-integration-dagster` is using?

So far I can make all tests passed provided that **Dagster version >= 1.2.0** with additional code in `conftest.py` for PoC when generating a job run.

```python
# To handle imports over different Dagster versions
if parse_version(DAGSTER_VERSION) <= parse_version("1.3.2"):
    from dagster.core.code_pointer import FileCodePointer
    from dagster.core.definitions.reconstruct import ReconstructableRepository
    from dagster.core.host_representation import ExternalPipelineOrigin
else:
    from dagster.core.host_representation import ExternalJobOrigin

if parse_version(DAGSTER_VERSION) <= parse_version("1.2.2"):
    from dagster.core.host_representation import InProcessRepositoryLocationOrigin
    from dagster.core.types.loadable_target_origin import LoadableTargetOrigin
else:
    from dagster.core.host_representation import InProcessCodeLocationOrigin
    from dagster.core.types.loadable_target_origin import LoadableTargetOrigin

...

def make_pipeline_run_with_external_pipeline_origin(
        repository_name: str,
):
    if parse_version(DAGSTER_VERSION) <= parse_version("1.2.2"):
        return DagsterRun(
            pipeline_name="test",
            execution_plan_snapshot_id="123",
            external_pipeline_origin=ExternalPipelineOrigin(
                external_repository_origin=ExternalRepositoryOrigin(
                    repository_location_origin=InProcessRepositoryLocationOrigin(
                        loadable_target_origin=LoadableTargetOrigin(
                            python_file="/openlineage/dagster/tests/test_pipelines/repo.py",
                        )
                    ),
                    repository_name=repository_name,
                ),
                pipeline_name="test",
            )
        )
    if parse_version(DAGSTER_VERSION) <= parse_version("1.3.2"):
        return DagsterRun(
            pipeline_name="test",
            execution_plan_snapshot_id="123",
            external_pipeline_origin=ExternalPipelineOrigin(
                external_repository_origin=ExternalRepositoryOrigin(
                    code_location_origin=InProcessCodeLocationOrigin(
                        loadable_target_origin=LoadableTargetOrigin(
                            python_file="/openlineage/dagster/tests/test_pipelines/repo.py",
                        )
                    ),
                    repository_name=repository_name,
                ),
                pipeline_name="test",
            )
        )
    return DagsterRun(
        job_name="test",
        execution_plan_snapshot_id="123",
        external_job_origin=ExternalJobOrigin(
            external_repository_origin=ExternalRepositoryOrigin(
                code_location_origin=InProcessCodeLocationOrigin(
                    loadable_target_origin=LoadableTargetOrigin(
                        python_file="/openlineage/dagster/tests/test_pipelines/repo.py",
                    )
                ),
                repository_name=repository_name,
            ),
            job_name="test",
        )
    )
```

Any thought in advance?

**Comment by JDarDagran** [comment]
@tsungchih It's installing `dagster-0.14.5` as defined [here](https://github.com/OpenLineage/OpenLineage/blob/main/integration/dagster/setup.py#L21). I think we can bump minimum version to 1.0.

**Comment by tsungchih** [comment]
@JDarDagran Thanks for your prompt reply. I have completed the fix to support Dagster version >= 1.0.0. However, the corresponding PR was tagged with additional labels. I think that is because I previously synced the fork and merged to my branch. I am not sure if that is the reason why some unnecessary tests were run against the dagster integration. I have tried to fix it but it didn't work. I may need help with the problem. Thanks!!
